plugins/crate-copy-azure/src/test/java/io/crate/copy/azure/AzureCopyIntegrationTest.java

```diff
@@ -0,0 +1,187 @@
+
+    private String containerUri;
+    private HttpServer httpServer;
+
+    /**
+     * OpenDAL uses shared singleton async tokio executor with configured number of threads.
+     * They are re-used throughout the app lifetime and cleaned up on executor disposal on node shutdown.
+     **/
+    public static class OpenDALFilter implements ThreadFilter {
+
+        @Override
+        public boolean reject(Thread t) {
+            // TODO: Used more reliable/less common pattern once https://github.com/apache/opendal/issues/5088 is implemented.
+            return t.getName().startsWith("Thread-");
+        }
+    }
+
+    @Override
+    protected Collection<Class<? extends Plugin>> nodePlugins() {
+        var plugins = new ArrayList<>(super.nodePlugins());
+        plugins.add(AzureCopyPlugin.class);
+        return plugins;
+    }
+
+    @Override
+    @Before
+    public void setUp() throws Exception {
+        super.setUp();
+        var handler = new AzureHttpHandler(CONTAINER_NAME, true);
+        httpServer = HttpServer.create(new InetSocketAddress(AZURE_ACCOUNT + "." + "localhost", 0), 0);
+        httpServer.createContext("/" + CONTAINER_NAME , handler);
+        httpServer.start();
+        InetSocketAddress address = httpServer.getAddress();
+
+        String host = String.format(
+            Locale.ENGLISH,
+            "%s:%d",
+            address.getAddress().getCanonicalHostName(),
+            address.getPort()
+        );
+        containerUri = String.format(
+            Locale.ENGLISH,
+            "az://%s.%s/%s",
+            AZURE_ACCOUNT,
+            host,
+            CONTAINER_NAME
+        );
+    }
+
+    @Override
+    @After
+    public void tearDown() throws Exception {
+        super.tearDown();
+        httpServer.stop(1);
+    }
+
+    @Test
+    public void test_copy_to_and_copy_from_azure_blob_storage_auth_via_key() throws IOException, InterruptedException {
+        execute("CREATE TABLE source (x int)");
+        execute("INSERT INTO source(x) values (1), (2), (3)");
+        execute("REFRESH TABLE source");
+
+        String fullURI = containerUri + "/dir/dir2";
+        execute("""
+            COPY source TO DIRECTORY ?
+            WITH (
+                protocol = 'http',
+                key = ?
+            )
+            """,
+            new Object[]{fullURI, AZURE_KEY}
+        );
+
+        execute("CREATE TABLE target (x int)");
+        execute("""
+            COPY target FROM ?
+            WITH (
+                protocol = 'http',
+                key = ?
+            )
+            """,
+            new Object[]{fullURI + "/*", AZURE_KEY}
+        );
+
+        execute("REFRESH TABLE target");
+        execute("select x from target order by x");
+        assertThat(response).hasRows("1", "2", "3");
+    }
+
+    @Test
+    public void test_copy_to_and_copy_from_azure_blob_storage_auth_via_token() throws IOException, InterruptedException {
```

Comment:
As the mocked HTTP Handler doesn't care about any parameter given (like if you pass in a `key` or `sas_token`) what value does this test add?

Reply:
Another argument for keeping it - only this test helped to find a bug in the parsing with URI - I forgot to add port.

https://github.com/crate/crate/pull/16631/commits/c24dfb29a69deaa24993b884549b4d0570616349